content/blog/2025-03-4-go-wrapper-in-public-preview.md

```diff
@@ -0,0 +1,410 @@
+authors= [ "niharikabhavaraju"]
++++
+
+Valkey-Glide is pleased to announce the public preview release of the GLIDE Go wrapper. This exciting addition to the Valkey ecosystem brings the power and reliability of Valkey to Go developers with a clean, intuitive API designed for performance and developer productivity.
+
+## Getting Started in Minutes
+
+The Valkey GLIDE Go wrapper provides a seamless experience for Go developers to interact with Valkey servers. With support for both standalone and cluster deployments, it's designed to be easy to integrate into your existing Go applications.
+
+### Installation is straightforward
+
+Add Valkey GLIDE to your project with just two commands:
+
+```bash
+go get github.com/valkey-io/valkey-glide/go
+go mod tidy
+```
+
+Then, connect to your Valkey server:
+
+### Standalone Mode Connection Setup
+
+For single-server deployments, the setup is simple:
+
+```go
+package main
+
+import (
+    "fmt"
+    "github.com/valkey-io/valkey-glide/go/api"
+)
+
+func main() {
+    // Connect to a standalone Valkey server
+    config := api.NewGlideClientConfiguration().
+        WithAddress(&api.NodeAddress{Host: "localhost", Port: 6379})
+    
+    client, err := api.NewGlideClient(config)
+    if err != nil {
+        fmt.Println("Error:", err)
+        return
+    }
+    defer client.Close()
+    
+    // Test the connection
+    result, err := client.Ping()
+    if err != nil {
+        fmt.Println("Error:", err)
+        return
+    }
+    fmt.Println(result) // PONG
+    
+    // Store and retrieve a value
+    client.Set("hello", "valkey")
+    value, _ := client.Get("hello")
+    fmt.Println(value) // valkey
+}
+```
+
+### Cluster Mode Connection Setup
+
+Need to work with a Valkey cluster?
+
+Just as easy! The Go wrapper automatically discovers your entire cluster topology from a single seed node:
+
+```go
+package main
+
+import (
+    "fmt"
+    "github.com/valkey-io/valkey-glide/go/api"
+)
+
+func main() {
+    host := "localhost"
+    port := 7001
+    
+    // Connect to a Valkey cluster through any node
+    config := api.NewGlideClusterClientConfiguration().
+        WithAddress(&api.NodeAddress{Host: host, Port: port})
+    
+    client, err := api.NewGlideClusterClient(config)
+    if err != nil {
+        fmt.Println("There was an error: ", err)
+        return
+    }
+    
+    res, err := client.Ping()
+    if err != nil {
+        fmt.Println("There was an error: ", err)
+        return
+    }
+    fmt.Println(res) // PONG
+    client.Close()
+}
+```
+
+## Why You Should Be Excited
+
+The Go wrapper extends Valkey GLIDE (General Language Independent Driver for the Enterprise) to the Go community, offering a robust, production-ready client that's built on the battle-tested Rust core. This client library is a thoughtfully designed experience for Go developers who need reliable, high-performance data access.
+
+## Key Features You'll Love
+
+### Smart Cluster Support with Automatic Topology Discovery
```

Comment:
This is a bit duplicative of the earlier section where you say it supports cluster topology with a single seed node. I would suggest just keeping one.